experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -467,13 +510,33 @@ export function applyTypesFromContext(
 		const fields: Map<FieldKey, MapTree[]> = new Map();
 		for (const key of fieldKeysFromData(data)) {
 			assert(!fields.has(key), 0x6b3 /* Keys should not be duplicated */);
-			const childSchema = getFieldSchema(key, schemaData, schema);
-			const children = applyFieldTypesFromContext(schemaData, childSchema, data[key]);
+			const childSchema = getFieldSchema(key, context.schema, schema);
+			const children = applyFieldTypesFromContext(context, childSchema, data[key]);
+
 			if (children.length > 0) {
 				fields.set(key, children);
 			}
 		}
 
+		// Generate and set data for global keys that have value fields.
+		if (context.schema.globalFieldSchema !== undefined) {
```

Comment:
there is no need to make this specific to global fields. It could apply to any field (local or global)

Also this if should never fail: globalFieldSchema is a member of the SchemaDataAndPolicy and is now allowed to be undefined so testing it for undefined should be pointless (unless we expect the runtime data to not match the type system, and in general we don't do that or having types is pointless).